libraries/chain/include/eosio/chain/chain_config.hpp

```diff
@@ -184,6 +184,67 @@ struct chain_config_v1 : chain_config_v0 {
    }
 };
 
+/**
+ * @brief v2 Producer-voted blockchain configuration parameters
+ *
+ * If Adding new parameters create chain_config_v[n] class instead of adding
+ * new parameters to v2, v1 or v0. This is needed for snapshots backward compatibility
+ */
+struct chain_config_v2 : chain_config_v1 {
+   using Base = chain_config_v1;
+
+   uint32_t   max_sync_call_depth = config::default_max_sync_call_depth;            ///< size limit for sync call depth
+   uint32_t   max_sync_call_data_size = config::default_max_sync_call_data_size;    ///< size limit for sync call data: input and return value respectively
+
+   //order must match parameters as ids are used in serialization
+   enum {
+     max_sync_call_depth_id = Base::PARAMS_COUNT,
+     max_sync_call_data_size_id,
+     PARAMS_COUNT
+   };
+
+   inline const Base& base() const {
+      return static_cast<const Base&>(*this);
+   }
+
+   void validate() const;
+
+   template<typename Stream>
+   friend Stream& operator << ( Stream& out, const chain_config_v2& c ) {
+      return c.log(out) << "\n";
```

Comment:
Should this have something else? v1 is quite different,
https://github.com/AntelopeIO/spring/blob/ea867bfee0890c49d4ca262f9927ee6dadf05958/libraries/chain/include/eosio/chain/chain_config.hpp#L182-L184

Reply:
Do I miss something of this comment? `Stream& log(Stream& out) const` is for `v1`'s `log` function, while `<<` is for v2's `<<` operator.

Reply:
Looks good I got mixed up